webroot/js/modules/Toolbar.js

```diff
@@ -0,0 +1,257 @@
+    window.parent.postMessage(state, window.location.origin);
+  }
+
+  hide() {
+    this.$container.removeClass('is-active');
+    this.currentPanelButton().removeClass('is-active');
+    const that = this;
+
+    // Shrink the iframe back to 40px height
+    setTimeout(() => {
+      window.parent.postMessage(that.state(), window.location.origin);
+    }, 250);
+  }
+
+  isExpanded() {
+    return this.$container.hasClass('is-active');
+  }
+
+  // ========== STATES ==========
+
+  state() {
+    return this.states[this.currentState];
+  }
+
+  nextState() {
+    this.currentState++;
+    if (this.currentState === this.states.length) {
+      this.currentState = 0;
+    }
+    this.saveState();
+    return this.state();
+  }
+
+  saveState() {
+    if (!this.isLocalStorageAvailable) {
+      return;
+    }
+    window.localStorage.setItem('toolbar_state', this.currentState);
+  }
+
+  loadState() {
+    if (this.isLocalStorageAvailable) {
+      let old = window.localStorage.getItem('toolbar_state');
+      if (!old) {
+        old = '0';
+      }
+      if (old === '0') {
+        this.hide();
+      }
+      if (old === '1') {
+        this.toggle();
+      }
+    }
+  }
+
+  // ========== UPDATE ELEMENTES DEPENDING ON STATE ==========
+
+  updateToolbarState(state) {
+    if (state === 'toolbar') {
+      this.$body.addClass('is-active');
+    }
+    if (state === 'collapse') {
+      this.$body.removeClass('is-active');
+    }
+  }
+
+  updateButtons(state) {
+    if (state === 'toolbar') {
+      this.$panelButtons.show();
+    }
+    if (state === 'collapse') {
+      this.$panelButtons.hide();
+    }
+  }
+
+  // ========== PANELS ==========
+
+  currentPanelButton() {
+    return this.$body.find(`[data-id="${this.currentPanel()}"]`);
+  }
+
+  currentPanel() {
+    return this.currentPanelId;
+  }
+
+  /**
+     * Responsible for loading a specific panel
+     *
+     * @param id The panel id
+     * @param panelType The panel type used for global custom events
+     */
+  loadPanel(id, panelType) {
+    if (id === undefined) {
+      return;
+    }
+
+    const url = `${this.baseUrl}debug-kit/panels/view/${id}`;
+    const contentArea = this.$container.find('.c-panel-content-container__content');
+    const that = this;
+    const loader = $('.o-loader');
+
+    const timer = setTimeout(() => {
+      loader.addClass('is-loading');
+    }, 500);
+
+    this.currentPanelId = id;
+    window.parent.postMessage('expand', window.location.origin);
+
+    $.get(url, (response) => {
+      clearTimeout(timer);
+      loader.removeClass('is-loading');
+
+      // Slide panel into place
+      that.$container.addClass('is-active');
+      contentArea.html(response);
+
+      // This initializes the panel specific JS logic (if there is any)
+      document.dispatchEvent(new CustomEvent('initPanel', { detail: `panel${panelType}` }));
+      that.bindNeatArray();
+      that.bindDebugBlock();
+    });
+  }
+
+  // ???????
+  bindNeatArray() {
```

Comment:
This can be removed.

Reply:
So can I remove the `ToolbarHelper::makeNeatArray()` method as well?

Reply:
Yes. All of that has been replaced with newer features in `Debugger`.